lib/daru/io/importers/excelx.rb

```diff
@@ -0,0 +1,132 @@
+        #
+        # @example Importing from a local file
+        #   path  = 'spec/fixtures/excelx/Stock-counts-sheet.xlsx'
+        #   sheet = 'Example Stock Counts'
+        #   df    = Daru::IO::Importers::XLSX.new(path, sheet: sheet).call
+        #   df
+        #
+        #   #=> <Daru::DataFrame(15x7)>
+        #              Status Stock coun  Item code        New Descriptio Stock coun Offset G/L
+        #        0          H          1        nil        nil New stock  2014-08-01        nil
+        #        1        nil          1  IND300654          2 New stock  2014-08-01      51035
+        #        2        nil          1   IND43201          5 New stock  2014-08-01      51035
+        #        3        nil          1   OUT30045          3 New stock  2014-08-01      51035
+        #       ...       ...        ...     ...           ...     ...       ...           ...
+        #
+        # @example Importing from a local file without headers
+        #   path  = 'spec/fixtures/excelx/Stock-counts-sheet.xlsx'
+        #   sheet = 'Example Stock Counts'
+        #   df    = Daru::IO::Importers::XLSX.new(path, sheet: sheet, headers: false).call
+        #   df
+        #
+        #   #=> <Daru::DataFrame(16x7)>
+        #                   0           1          2          3          4          5        6
+        #        0      Status Stock coun  Item code        New Descriptio Stock coun Offset G/L
+        #        1          H          1        nil        nil New stock  2014-08-01        nil
+        #        2        nil          1  IND300654          2 New stock  2014-08-01      51035
+        #        3        nil          1   IND43201          5 New stock  2014-08-01      51035
+        #        4        nil          1   OUT30045          3 New stock  2014-08-01      51035
+        #       ...       ...        ...     ...           ...     ...       ...           ...
+        #
+        # @example Importing from a remote URL
+        #   path = 'https://www.exact.com/uk/images/downloads/getting-started-excel-sheets/Stock-counts-sheet.xlsx'
+        #   sheet = 'Example Stock Counts'
+        #   df    = Daru::IO::Importers::XLSX.new(path, sheet: sheet).call
+        #   df
+        #
+        #   #=> <Daru::DataFrame(15x7)>
+        #              Status Stock coun  Item code        New Descriptio Stock coun Offset G/L
+        #        0          H          1        nil        nil New stock  2014-08-01        nil
+        #        1        nil          1  IND300654          2 New stock  2014-08-01      51035
+        #        2        nil          1   IND43201          5 New stock  2014-08-01      51035
+        #        3        nil          1   OUT30045          3 New stock  2014-08-01      51035
+        #       ...       ...        ...     ...           ...     ...       ...           ...
+        def initialize(path, sheet: 0, order: true, index: false, skiprows: 0, skipcols: 0)
+          optional_gem 'roo', '~> 2.7.0'
```

Comment:
In fact, this can be metaprogrammed. Like this:

```ruby
class Excelx < Base
  requires_gem 'roo', '~> 2.7.0' # says that on real creation of object the check should be done
```

Another thought is you probably should not specify dependencies version so narrow for general-usage gem (which leads to idea of testing with multiple dependencies versions... So we should postpone this, just don't forget)